cli/file_watcher.rs

```diff
@@ -15,57 +13,42 @@ use notify::Error as NotifyError;
 use notify::RecommendedWatcher;
 use notify::RecursiveMode;
 use notify::Watcher;
-use pin_project::pin_project;
 use std::collections::HashSet;
 use std::path::PathBuf;
-use std::pin::Pin;
 use std::sync::Arc;
-use std::task::Context;
-use std::task::Poll;
 use std::time::Duration;
-use tokio::pin;
 use tokio::select;
+use tokio::sync::mpsc;
 use tokio::time::sleep;
-use tokio::time::Instant;
-use tokio::time::Sleep;
 
 const DEBOUNCE_INTERVAL: Duration = Duration::from_millis(200);
 
-#[pin_project(project = DebounceProjection)]
-struct Debounce {
-  #[pin]
-  timer: Sleep,
-  changed_paths: Arc<Mutex<HashSet<PathBuf>>>,
+struct DebouncedReceiver {
+  receiver: mpsc::UnboundedReceiver<Vec<PathBuf>>,
 }
 
-impl Debounce {
-  fn new() -> Self {
-    Self {
-      timer: sleep(DEBOUNCE_INTERVAL),
-      changed_paths: Arc::new(Mutex::new(HashSet::new())),
-    }
+impl DebouncedReceiver {
+  fn new_with_sender() -> (Arc<mpsc::UnboundedSender<Vec<PathBuf>>>, Self) {
+    let (sender, receiver) = mpsc::unbounded_channel();
+    (Arc::new(sender), Self { receiver })
   }
-}
 
-impl Stream for Debounce {
-  type Item = Vec<PathBuf>;
-
-  /// Note that this never returns `Poll::Ready(None)`, which means that the
-  /// file watcher will be alive until the Deno process is terminated.
-  fn poll_next(
```

Comment:
IMO, I think we should try to avoid implementing this trait if able (maybe only mission critical stuff). I think the new debounced channel is easier to understand.